TemplateMethod/Program.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Epam.NetMentoring.Adapter.TemplateMethod
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            var template = new Template
+            {
+                Name = "Test",
+                CustomerName = "TestCustomer",
+                Email = new List<string>() { "test@test.com" },
+                isRegistered = true,
+                MsgContent = "Hello",
+                User = new List<string>() { "TestUser" }
```

Comment:
I guess user should be separated entity, because as per requirements - we need to check does user have access to template, so template is class and user is also class:
`
var user = new User(...)
var template = new Template(...);
validator.Validate(user, template)
`

Reply:
ok

Reply:
ok